src/simulation/decision_model_simulation_results.jl

```diff
@@ -263,10 +263,7 @@ function _read_results(
     isempty(result_keys) &&
         return Dict{OptimizationContainerKey, ResultsByTime{DenseAxisArray{Float64, 2}}}()
 
-    if store === nothing && res.store !== nothing
-        # In this case we have an InMemorySimulationStore.
-        store = res.store
-    end
+    _store = _try_resolve_store(store, res.store)
```

Comment:
There is another location in emulation_model_simulation_results.jl that needs to be changed.